services/AttendanceService.ts

```diff
@@ -53,4 +55,27 @@ export default class AttendanceService {
       return attendance.getPublicAttendance();
     });
   }
+
+  public async addEventFeedback(feedback: string[], uuid: Uuid, user: UserModel): Promise<PublicAttendance> {
+    return this.transactions.readWrite(async (txn) => {
+      const attendanceRepository = Repositories.attendance(txn);
+      const event = await Repositories.event(txn).findByUuid(uuid);
+      if (!event) throw new NotFoundError('Event not found');
+      const hasAttended = await attendanceRepository.hasUserAttendedEvent(user, event);
+      if (!hasAttended) throw new UserError('You cannot provide feedback for an event you didn\'t attend');
+      const hasProvidedFeedback = await attendanceRepository.hasUserProvidedEventFeedback(user, event);
+      if (hasProvidedFeedback) throw new UserError('You cannot submit feedback for this event more than once');
+      const eventStart = moment(event.start).valueOf();
+      const twoDaysPastEventEnd = moment(event.end).add(2, 'days').valueOf();
+      if (moment.now() < eventStart) throw new UserError('You cannot submit feedback until the event has started');
+      if (moment.now() > twoDaysPastEventEnd) {
+        throw new UserError('You cannot submit feedback past 2 days of the event ending');
```

Comment:
for this and the other error messages: rephrase as "you must submit within 2 days" or "you must attend before feedback"